### PR TITLE
Feature/graphql vendors refactor

### PR DESCRIPTION
**Pull request by lafbelmonte**
-Updated use case return to boolean for graphql mutations
-Updated all tests to adopt new use case
-Used chance to mock name
-Change type of id to ID! in type definitions

### CODE REVIEW

File: koa-crud/src/graphql/typeDefs.ts [hunk 1]
```diff
@@ -2,27 +2,17 @@ import { gql } from 'apollo-server-koa';
...
-  }
-  type ArrayOfVendorReturn {
-    message: String
-    data: [Vendor]
-  }
-  type VendorReturn {
-    message: String
-    data: Vendor
+    createVendor(name: String, type: String): Boolean
+    updateVendor(id: ID!, name: String, type: String): Boolean
```
Comment (ralphcasipe1) [inline koa-crud/src/graphql/typeDefs.ts]: please add input for `createVendor` and `updateVendor`

e.g.

```gql
input CreateVendorInput {
  name: String!
  type: VendorType!
}

type Mutation {
   createVendor(input: CreateVendorInput!): Boolean
}
```

File: koa-crud/src/data-access/mongoose/vendors/actions.ts [hunk 2]
```diff
@@ -12,11 +12,11 @@ const actions = ({ Vendor }): VendorStore => {
   }
 
   async function selectAllVendors() {
-    return Vendor.find().lean();
+    return Vendor.find().lean({virtuals: true});
   }
 
   async function selectOneVendorByFilters(filters) {
-    return Vendor.findOne({ ...filters }).lean();
+    return Vendor.findOne({ ...filters }).lean({virtuals: true});
```
Comment (ralphcasipe1) [inline koa-crud/src/data-access/mongoose/vendors/actions.ts]: If you are only using `filters` here
```ts
{ ...filters }
```

You can also apply it directly
```ts
    return Vendor.findOne(filters).lean({virtuals: true});

```

File: koa-crud/src/graphql/typeDefs.ts [hunk 3]
```diff
@@ -2,27 +2,17 @@ import { gql } from 'apollo-server-koa';
...
+    deleteVendor(id: ID!): Boolean
   }
+
   type Vendor {
-    _id: ID!
+    id: ID!
     name: String
     type: String
```
Comment (ralphcasipe1) [inline koa-crud/src/graphql/typeDefs.ts:33]: Add a enum `VendorType`
```gql
enum VendorType {
  SEAMLESS
  TRASFER
}
```

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]
Next time please, make sure to have only one intent in a feature. As much as possible name your feature branch base on an intent for example:
- `feature/adopt-graphql`
- `feature/update-test-use-case`

This will help reviewers focus on one thing and minimizes mental mapping

File: koa-crud/src/graphql/typeDefs.ts (hunk 1 above)
Comment (lafbelmonte) [reply koa-crud/src/graphql/typeDefs.ts]: noted sir

File: koa-crud/src/data-access/mongoose/vendors/actions.ts (hunk 2 above)
Comment (lafbelmonte) [reply koa-crud/src/data-access/mongoose/vendors/actions.ts]: noted sir

File: koa-crud/src/graphql/typeDefs.ts (hunk 3 above)
Comment (lafbelmonte) [reply koa-crud/src/graphql/typeDefs.ts:33]: noted sir

**Review by ralphcasipe1 — APPROVED** [review]

**Pull request merged**